pystac/extensions/eo.py

```diff
@@ -274,7 +275,9 @@ def band_description(common_name: str) -> Optional[str]:
 
 
 class EOExtension(
-    Generic[T], PropertiesExtension, ExtensionManagementMixin[pystac.Item]
+    Generic[T],
```

Comment:
I had originally added `pystac.Collection` to the `Generic[T]` type param as well, but I think that was incorrect -- the `T` param there I think refers to the specific types of entities that this extension applies directly to, and collections aren't one of those

Reply:
That's correct, the `T` type should only apply to objects that can be extended by this extension (in this case `Asset` and `Item`) and the generic argument(s) to `ExtensionManagementMixin` cover any object that can contain the extension URI in its `stac_extensions` list (in this case `Item` and `Collection`).